### ISSUE TITLE
Error adding a new card: You passed an empty string for 'billing_details[address][country]' ...

### ISSUE THREAD

**Issue by Ariandr** [stripe/stripe-ios#1752]
## Summary
I migrated from SDK v.19.4.0 to 21.2.0.

While using 21.2.0 I get this error. It shows this message: 'You passed an empty string for 'billing_details[address][country]' ... . Please, look at the screenshot.

When using 19.4.0 I didn't get this error and everything worked fine with the same logic, the code was changed only to make the project compile.

I set in App Delegate:
`STPPaymentConfiguration.shared.requiredBillingAddressFields = .none`
Apart from that, I don't specify that any other billing details must be collected in any place in the app as far as I can see.

<img width="483" alt="Screenshot 2021-01-21 at 15 27 44" src="https://user-images.githubusercontent.com/13591333/105359273-edd6d680-5bff-11eb-92e9-1360f17e4f3b.png">

## Code to reproduce
I initialize customer and payment contexts in this way not specifying anything extra.
```
self.customerContext = STPCustomerContext(keyProvider: StripeAPIClient.shared)
let configuration = STPPaymentConfiguration.shared
self.paymentContext = STPPaymentContext(customerContext: customerContext, configuration: configuration, theme: STPTheme.defaultTheme)
```

## iOS version
14.3

## Installation method
Cocoapods, 1.10.1

## SDK version
21.2.0

## Question:
How do I solve it? Should I set some properties somewhere?
I didn't change behavior in any way after installing 21.2.0. I just implemented necessary migration steps for the project to compile. 

Thanks in advance!

**Comment by Ariandr** [comment]
Hi @davidme-stripe 
Have you had a chance to look into it?

It's a blocking issue which persists.
By the way, in this example I used a test card 4242 4242 4242 4242.

**Comment by davidme-stripe** [comment]
Hi, sorry for the delay! This is a combination of a bug on our end and an issue with the iOS Simulator.

1) The iOS Simulator sometimes returns `nil` as the current locale's country code. You can fix this by going into the Simulator's Settings -> General -> Language & Region -> Region and setting a country code (even if one is already set, just re-set the one that's selected).

2) Our SDK doesn't correctly handle a `nil` current locale in all situations. We'll fix that in an update soon. I don't think users would hit this in the wild, as real devices should always have a valid locale.

**Comment by Ariandr** [comment]
Hi @davidme-stripe 
Thank you!
I will try tomorrow and get back to you.

**Comment by Ariandr** [comment]
Hi @davidme-stripe 
I tried resetting the region and it works now on the Simulator, thanks.

I'm closing the issue, maybe it will be useful for folks if they face the same issue.

**Comment by davidme-stripe** [comment]
This was fixed in [21.2.1](https://github.com/stripe/stripe-ios/releases/tag/21.2.1), so it should now work again with invalid regions.
